File: js/anagram2.js
exports.anagramsFor = function(word, listOfWords) {
    
    let answer = []
    //sorting the word parameter alphabetically
    sortWord = word.split('').sort().join('')

    //sorting each word in listOfWords alphabetically
    sortListOfWords = listOfWords.map(elem => elem.split('').sort().join(''))

    //iterating through the sortListOfWords and check if each element == sortWord
    for (let i = 0; i < sortListOfWords.length; i++) {
        if (sortListOfWords[i] === sortWord) {
            answer.push(listOfWords[i])
        }
    } return answer

};

File: python/anagram2.py
# Don't forget to run the tests (and create some of your own)

from ast import JoinedStr
from operator import indexOf


def anagrams_for(word, list_of_words):
	answer = []
	sort_word = ("".join((sorted(word))))


	for i in range(0, len(list_of_words), 1) :
		idx_word = ("".join(sorted(list(list_of_words[i]))))
		if idx_word == sort_word :
			answer.append(list_of_words[i])
	
	return answer

